var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -94,7 +94,10 @@ class Mpich(AutotoolsPackage):
 '-Wl,-flat_namespace'.'''
     )
 
-    provides('mpi@:3.1')
+    variant('cuda', default=False, when="@3.4: device=ch4", description="Enable CUDA support")
```

Comment:
@becker33 what's the expected behavior when the derived class overrides a variant and makes it conditional but the base class defines it unconditionally? Right now `spack spec mpich@:3.0 device=ch3 +cuda` concretizes fine, is that a bug?

Reply:
@raffenet sorry for all the back and forth in this PR delaying the merge. Two conflicts `conflict('+cuda', when='@:3.3')` and `conflict('+cuda', when='device=ch3')` is the better option until the above bug (?) is fixed :(

Reply:
This looks good to me.